src/test/java/io/debezium/connector/vitess/VitessSchemaMigrationIT.java

```diff
@@ -0,0 +1,114 @@
+        final VitessDatabaseSchema vitessDatabaseSchema = new VitessDatabaseSchema(
+                conf, SchemaNameAdjuster.create(), VitessTopicSelector.defaultSelector(conf));
+        AtomicReference<Throwable> error = new AtomicReference<>();
+        try (VitessReplicationConnection connection = new VitessReplicationConnection(conf, vitessDatabaseSchema)) {
+            Vgtid startingVgtid = Vgtid.of(
+                    Binlogdata.VGtid.newBuilder()
+                            .addShardGtids(
+                                    Binlogdata.ShardGtid.newBuilder()
+                                            .setKeyspace(conf.getKeyspace())
+                                            .setShard(conf.getShard())
+                                            .setGtid("current")
+                                            .build())
+                            .build());
+
+            BlockingQueue<MessageAndVgtid> consumedMessages = new ArrayBlockingQueue<>(100);
+            AtomicBoolean started = new AtomicBoolean(false);
+            connection.startStreaming(
+                    startingVgtid,
+                    (message, vgtid, isLastRowEventOfTransaction) -> {
+                        if (!started.get()) {
+                            started.set(true);
+                        }
+                        consumedMessages.add(new MessageAndVgtid(message, vgtid));
+                    },
+                    error);
+            Awaitility
+                    .await()
+                    .atMost(Duration.ofSeconds(TestHelper.waitTimeForRecords()))
+                    .until(started::get);
+            consumedMessages.clear();
+            // OnlineDDL is async, hence we issue the command and wait until it finishes
+            String ddlId = TestHelper.applySchema("ALTER TABLE t1 ADD COLUMN name varchar(64)", TestHelper.TEST_UNSHARDED_KEYSPACE);
+            Awaitility
+                    .await()
+                    .atMost(Duration.ofSeconds(TestHelper.waitTimeForRecords()))
+                    .until(() -> {
+                        Thread.sleep(1000);
+                        return TestHelper.checkOnlineDDL(TestHelper.TEST_UNSHARDED_KEYSPACE, ddlId);
+                    });
+            TestHelper.execute("UPDATE t1 SET name='abc' WHERE id=1");
+            MessageAndVgtid message = TestHelper.awaitOperation(
+                    TestHelper.waitTimeForRecords(),
+                    SECONDS,
+                    "UPDATE",
+                    () -> {
+                        try {
+                            return consumedMessages.poll(SECONDS.toMillis(5), TimeUnit.MILLISECONDS);
+                        }
+                        catch (InterruptedException e) {
+                            return null;
+                        }
+                    });
+            ReplicationMessage dml = message.getMessage();
```

Comment:
So I'm a bit confused about this test; I would rather have expected a new test method added to `VitessConnectorIT`, which asserts the actual `SourceRecord` emitted by the connector, including the schema of the new field. I fear with the currently proposed test we're not making sure that the functionality actually works as intended.

Reply:
Good point, I added another test case in `VitessConnectorIT` and also move this one into `VitessReplicationConnectionIT` - lmk if this makes sense to you :-)